neurodamus/utils/memory.py

```diff
@@ -480,56 +447,46 @@ def distribute_cells(self, num_ranks, batch_size=10) -> (dict, dict):
 
         self.validate_inputs_distribute(num_ranks, batch_size)
 
-        # Multiply the average number of synapses per cell by 2.0
-        # This is done since the biggest memory load for a synapse is 2.0 kB and at this point in
-        # the code we have lost the information on whether they are excitatory or inhibitory
-        # so we just take the biggest value to be safe. (the difference between the two is minimal)
-        average_syns_mem_per_cell = {k: v * 2.0 for k, v in self.average_syns_per_cell.items()}
-
-        # Prepare a list of tuples (cell_id, memory_load)
-        # We sum the memory load of the cell type and the average number of synapses per cell
-        def generate_cells(metype_gids):
-            for cell_type, gids in metype_gids.items():
-                memory_usage = (self.metype_memory[cell_type] +
-                                average_syns_mem_per_cell[cell_type])
-                for gid in gids:
-                    yield gid, memory_usage
-
```

Comment:
Removing this generator is probably gonna cause a massive pain when we will merge/rebase this with the multi-cycle MR since I still use these structures.

Reply:
If not removed it would have been modified quite a bit, so conflicts are unavoidable. And we were kind of duplicating some logic, the new code is simpler